platform/viewer/src/routes/Mode/Mode.tsx

```diff
@@ -8,6 +8,52 @@ import { DragAndDropProvider, ImageViewerProvider } from '@ohif/ui';
 import { useQuery, useSearchParams } from '@hooks';
 import ViewportGrid from '@components/ViewportGrid';
 import Compose from './Compose';
+import { StudyMetadata } from 'platform/core/src/Types';
+
+const compare = (a, b, def = 0): number => {
+  if (a === b) return def;
+  if (a < b) return 1;
+  return -1;
+};
+
+/** The studies from display sets gets the studies in whatever
+ * order the display sets are in.
+ */
+const getStudiesfromDisplaySets = (
+  metadataStore,
+  displaysets
+): StudyMetadata => {
+  const studyMap = {};
+
+  const ret = displaySets.reduce((prev, curr) => {
+    const { StudyInstanceUID } = curr;
+    if (!studyMap[StudyInstanceUID]) {
+      const study = DicomMetadataStore.getStudy(StudyInstanceUID);
+      studyMap[StudyInstanceUID] = study;
+      prev.push(study);
+    }
+    return prev;
+  }, []);
+  ret.sort((a, b) => {
+    return compare(
+      a.StudyDate,
+      b.StudyDate,
+      compare(a.StudyInstanceUID, b.StudyInstanceUID)
+    );
+  });
+  return ret;
+};
+
+/** The studies retrieve from the Uids is faster and gets the studies
+ * in the original order, as specified.
+ */
+const getStudiesFromUids = (
+  metadataStore,
+  studyUids: string[]
+): StudyMetadata[] => {
+  if (!studyUids?.length) return;
+  return studyUids.map(uid => DicomMetadataStore.getStudy(uid));
+};
```

Comment:
Can we put these two helpers inside a helper or utils folder? `Mode.tsx` is one of the most important files in the whole repo